solana/restaking/programs/restaking/src/lib.rs

```diff
@@ -96,47 +96,51 @@ pub mod restaking {
         // Mint receipt tokens
         token::mint_nft(ctx.accounts.into())?;
 
-        // Call Guest chain program to update the stake if the chain is initialized
-        if guest_chain_program_id.is_some() {
-            let validator_key = match service {
-                Service::GuestChain { validator } => validator,
-            };
-            let borrowed_chain_data =
-                ctx.remaining_accounts[0].data.try_borrow().unwrap();
-            let mut chain_data: &[u8] = &borrowed_chain_data;
-            let chain =
-                solana_ibc::chain::ChainData::try_deserialize(&mut chain_data)
-                    .unwrap();
-            let validator = chain
-                .validator(validator_key)
-                .map_err(|_| ErrorCodes::OperationNotAllowed)?;
-            let amount = validator.map_or(u128::from(amount), |val| {
-                u128::from(val.stake) + u128::from(amount)
-            });
-            validation::validate_remaining_accounts(
-                ctx.remaining_accounts,
-                &guest_chain_program_id.unwrap(),
-            )?;
-            core::mem::drop(borrowed_chain_data);
-            let bump = ctx.bumps.staking_params;
-            let seeds =
-                [STAKING_PARAMS_SEED, TEST_SEED, core::slice::from_ref(&bump)];
-            let seeds = seeds.as_ref();
-            let seeds = core::slice::from_ref(&seeds);
-            let cpi_accounts = SetStake {
-                sender: ctx.accounts.depositor.to_account_info(),
-                chain: ctx.remaining_accounts[0].clone(),
-                trie: ctx.remaining_accounts[1].clone(),
-                system_program: ctx.accounts.system_program.to_account_info(),
-                instruction: ctx.accounts.instruction.to_account_info(),
-            };
-            let cpi_program = ctx.remaining_accounts[2].clone();
-            let cpi_ctx =
-                CpiContext::new_with_signer(cpi_program, cpi_accounts, seeds);
-            solana_ibc::cpi::set_stake(cpi_ctx, validator_key, amount)?;
-        }
+        // Call Guest chain program to update the stake if the chain is
+        // initialised.
+        let guest_chain_program_id = match guest_chain_program_id {
+            Some(id) => id,
+            None => return Ok(()),
```

Comment:
why is it `Ok`?

Reply:
Hmm, it is what existing code does. @dhruvja, should we fail instead? It does look like we’re not recording the stake and just forget about the deposit when guest chain program is not set?

Reply:
so the `guest_chain_program_id` being `None` indicates that the vaults are only for depositing and not ready for delegating or withdrawing. It was done because we still didnt have the solana-ibc ready when we deployed restaking program. Now that it is set, we can prolly panic but i dont think it affects anyhow.

Reply:
SG.  If it doesn’t affect correctness, let’s keep it as is and we can change it in another PR if we so desire.